weather_server/show_weather/views.py

```diff
@@ -75,4 +76,22 @@ def csv_observation_request(request, station, start, end):
         )
     )
     response.seek(0)
-    return HttpResponse(response)
+    return HttpResponse(response, content_type="text/html")
+
+def png_observation_request(request, station, start, end):
+    """Return PNG of observations in requested time range."""
+    obs = Observation.objects.filter(station__station_id=station,
+                                     obs_date__gte=start,
+                                     obs_date__lte=end)
+
+    #TODO create n sub plots for all relevant parameter or merge them in 1 plot	
+    # TODO how to loop over all elements and extract name along the way?
+    obsdata = [o.temperature for o in obs]
```

Comment:
Here a generator expression would be more appropriate - it just saves the instructions for pulling out the data and yields it dynamically, so you don't end up passing around large data structures in memory. If I remember correctly, ``plt.plot`` will evaluate a generator expression as an iterable, consuming it to produce the data it needs.

Pulling the data out of the DB as a list isn't a bad thing, but it does requrie a lot of memory manipulation that we'll notice as soon as we're working with a lot of data.